test/functional/mempool_updatefromblock.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mempool descendants/ancestors information update.
+
+Test mempool update of transaction descendants/ancestors information (count, size)
+when transactions have been re-added from a disconnected block to the mempool.
+"""
+import time
+
+from decimal import Decimal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class MempoolUpdateFromBlockTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.extra_args = [['-limitdescendantsize=1000', '-limitancestorsize=1000']]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', end_address='', fee=Decimal(0.00100000)):
+        """Create a tournament (a type of directed graph) of transactions and use it for testing.
+
+        Keyword arguments:
+        size -- the order N of the tournament which is equal to the number of the created transactions
+        n_tx_to_mine -- the number of transaction that should be mined into a block
+
+        If all of the N created transactions tx[0]..tx[N-1] reside in the mempool,
+        the following holds:
+            the tx[K] transaction:
+            - has N-K descendants (including this one), and
+            - has K+1 ancestors (including this one)
+
+        More details: https://en.wikipedia.org/wiki/Tournament_(graph_theory)
```

Comment:
A tournament can be a direct cyclic graph? But utxo graph has to be a direct acyclic graph and should be forbid in the mempool, for preciseness, you may indicate it generates a subset of tournament.

Reply:
I think "acyclic" property is implied in this case :)

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18485#issuecomment-619077646).